public/js/index.js

```diff
@@ -128,250 +133,196 @@ function handleViewClick(lesson) {
+
+			const titleContainer = document.createElement('div');
+			titleContainer.classList.add('lesson-card-title-container');
+
+			const lessonTitle = document.createElement('h2');
+			lessonTitle.classList.add('lesson-card-title');
+			lessonTitle.textContent = title;
+
+			const lessonContent = document.createElement('div');
+			lessonContent.classList.add(
+				'lesson-card-content',
+				'ql-editor',
+				'ql-container',
+			);
+			lessonContent.innerHTML = content;
+
+			const lessonRemoveBtn = document.createElement('button');
+			lessonRemoveBtn.classList.add('button');
+			lessonRemoveBtn.id = 'delete';
+
+			const removeIcon = document.createElement('img');
+			removeIcon.setAttribute('alt', 'remove lesson icon');
+			removeIcon.setAttribute('src', './images/cancel-white.svg');
+			removeIcon.id = 'delete';
+
+			const editIcon = document.createElement('img');
+			editIcon.setAttribute('alt', 'edit lesson icon');
+			editIcon.setAttribute('src', './images/edit-white.svg');
+			editIcon.id = 'edit';
+
+			const lessonEditBtn = document.createElement('button');
+			lessonEditBtn.classList.add('button');
+			lessonEditBtn.id = 'edit';
+
+			const lessonViewBtn = document.createElement('button');
+			lessonViewBtn.textContent = 'VIEW LESSON';
+			lessonViewBtn.id = 'view';
+			lessonViewBtn.classList.add('button');
+
+			titleContainer.appendChild(lessonTitle);
+			lessonRemoveBtn.appendChild(removeIcon);
+			titleContainer.appendChild(lessonRemoveBtn);
+			lessonCard.appendChild(titleContainer);
+			lessonCard.appendChild(lessonContent);
+			buttonContainer.appendChild(lessonViewBtn);
+			lessonEditBtn.appendChild(editIcon);
+			buttonContainer.appendChild(lessonEditBtn);
+			lessonCard.appendChild(buttonContainer);
+			lessonsContainer.appendChild(lessonCard);
 		});
+	} else {
+		const noLessons = document.createElement('p');
+		noLessons.classList.add('no-lessons');
+
+		noLessons.textContent = 'No lessons :(';
+		lessonsContainer.appendChild(noLessons);
+	}
 }
 
-function addUser(user) {
+async function addUser(user) {
 	const { uid } = user;
+	const snapshot = await db.collection('users').where('uid', '==', uid).get();
+	const userExists = !snapshot.empty;
 
-	db.collection('users')
-		.where('uid', '==', uid)
-		.get()
-		.then((snapshot) => {
-			const userExists = !snapshot.empty;
-
-			if (userExists) {
-				console.log('user already exists');
-			} else {
-				console.log('user does not exist. Adding...');
-				db.collection('users')
-					.doc(uid)
-					.set({ uid: uid, lessons: [] })
-					.catch((error) => {
-						console.error('Error adding document: ', error);
-					});
-			}
-		});
+	if (userExists) {
+		console.log('user already exists');
+	} else {
+		console.log('user does not exist. Adding...');
+		db.collection('users')
+			.doc(uid)
+			.set({ uid: uid, lessons: [] })
+			.catch((error) => {
+				console.error('Error adding document: ', error);
+			});
+	}
 }
 
-function addLesson(e) {
-	e.preventDefault();
```

Comment:
Is there a reason why `e.preventDefault();` has been removed from the function? I can see you've moved it into an anonymous function on line 29.

Reply:
Yeah, so e.preventDefault() did not work when I refactored this function into async/await. So I looked it up and followed this guideline:

https://github.com/github/eslint-plugin-github/blob/main/docs/rules/async-preventdefault.md

Reply:
That makes sense. Good to know too, thanks for sharing!